cache/lru_cache.cc

```diff
@@ -338,81 +456,32 @@ bool LRUCacheShard::Release(Cache::Handle* handle, bool force_erase) {
 Status LRUCacheShard::Insert(const Slice& key, uint32_t hash, void* value,
                              size_t charge,
                              void (*deleter)(const Slice& key, void* value),
+                             Cache::CacheItemHelperCallback helper_cb,
                              Cache::Handle** handle, Cache::Priority priority) {
   // Allocate the memory here outside of the mutex
   // If the cache is full, we'll have to release it
   // It shouldn't happen very often though.
   LRUHandle* e = reinterpret_cast<LRUHandle*>(
       new char[sizeof(LRUHandle) - 1 + key.size()]);
-  Status s = Status::OK();
-  autovector<LRUHandle*> last_reference_list;
 
   e->value = value;
-  e->deleter = deleter;
+  e->flags = 0;
+  if (helper_cb) {
+    e->SetTieredCacheCompatible(true);
+    e->info_.helper_cb = helper_cb;
```

Comment:
Why we don't set e->info_.deleter here through `ShardedCache::DeletionCallback del_cb; (*info_.helper_cb)(nullptr, nullptr, &del_cb);`? Does it means, the call back function can be different when we call helper_cb at different time period?

Reply:
The reason is we also need callbacks for demoting/promoting the entry, in addition to deleting it. Instead of keeping track of all 3, just track the ```helper_cb```. It saves some metadata in ```LRUHandle```.